solr/core/src/java/org/apache/solr/search/SolrCache.java

```diff
@@ -174,4 +174,12 @@ default void close() throws IOException {
    * only on implementations that support it, it's a no-op otherwise.
    */
   void setMaxRamMB(int maxRamMB);
+
+  /**
+   * Check if this SolrCache supports async queries - relevant when attempting to cache nested or recursive results.
```

Comment:
IMO just keep it simple; don't specify why you might want it.  That reason may become obsolete.
Also that javadoc doesn't seem helpful for those not in-the-know.  I suggest calling this `isRecursionSupported()' and document it by pointing to ConcurrentHashMap.computeIfAbsent to say that it does *not* support it.